### ISSUE TITLE
Pass design time values to composite components

### ISSUE THREAD

**Issue by fawadasaurus** [candlecorp/wick#233]
**Is your feature request related to a problem? Please describe.**
I want to pass secrets or other design time configurations to composite components. For other components, those are sent using the `with` yaml as they can be used in the web assembly programming logic as needed.

**Propose a solution**
When a `with` value is passed to a composite component, we need a way to get the design time value passed to the composite component. It will likely need its own semantic like `$CONFIG.foo` for variable expansion.

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by fawadasaurus** [comment]
Here is an idea for a yaml that keeps the syntax consistent with the rust code:

`ctx.root_config.token` to access the `with` config.  This is consistent with the rust code.
```rust
let config: &RootConfig = ctx.root_config();
let token = config.token;
```

```yaml
kind: wick/component@v1
name: test_api
metadata:
  version: 0.0.1
  description: New HTTP Client wick component
  licenses:
    - Apache-2.0
resources:
  - name: HTTP_URL
    resource:
      kind: wick/resource/url@v1
      url: https://mydomain/lookup
component:
  kind: wick/component/http@v1
  with:
    - name: token
      type: string
  resource: HTTP_URL
  codec: Json
  operations:
    - name: geolocate
      inputs:
        - name: ip
          type: string
      method: Post
      codec: Json
      headers:
        "Authorization": ["Bearer {{ctx.root_config.token}}"],
        "Content-Type": ["application/json"]
      body:
        ip: "{{ip}}"
```

**Comment by jsoverson** [comment]
Got things working for with-config. How's this feel so far:

```yaml
---
name: 'test'
kind: wick/component@v1
metadata:
  version: '0.0.2'
component:
  kind: wick/component/composite@v1
  with:
    - name: component_config_greeting
      type: string
  operations:
    - name: test
      with:
        - name: op_config_name
          type: string
      uses:
        - name: GREET
          operation: self::greet
          with:
            greeting: '{{ ctx.root.component_config_greeting }}, {{ ctx.config.op_config_name }}! '
      inputs:
        - name: input
          type: string
      outputs:
        - name: output
          type: string
      flow:
        - <>.input -> GREET.input
        - GREET.output -> <>.output
    - name: greet
      with:
        - name: greeting
          type: string
      uses:
        - name: GREETING
          operation: core::sender
          with:
            output: '{{ ctx.config.greeting }}'
      flow:
        - <>.input -> test::concat[a].right
        - GREETING.output -> a.left
        - a.output -> <>.output

```

**Comment by fawadasaurus** [comment]
in rust, its accessed via `root_config()` so it would keep the consistency by calling it `root_config`
`ctx.root_config.component_config_greeting`

**Comment by jsoverson** [comment]
Addressed via #319
